### PR TITLE
Allow several tempest runs with different config

### PR DESCRIPTION
**Pull request by AurelienLourot**
This deprecates:

    configure:
      - zaza.openstack.charm_tests.tempest.setup.render_tempest_config_keystone_v3
    tests:
      - zaza.openstack.charm_tests.tempest.tests.TempestTest

by allowing multiple runs with different config:

    tests:
      - zaza.openstack.charm_tests.tempest.tests.TempestTestWithKeystoneV2
      - zaza.openstack.charm_tests.openstack_upgrade.tests.OpenStackUpgradeTestsByOption
      - zaza.openstack.charm_tests.tempest.tests.TempestTestWithKeystoneV3

### CODE REVIEW

**Comment by AurelienLourot** [comment]
Validating this change as a func-test-pr for the charm-glance gate:
* validating backward compatibility [here](https://review.opendev.org/c/openstack/charm-glance/+/809187)
* validating new test class [here](https://review.opendev.org/c/openstack/charm-glance/+/809189)

File: zaza/openstack/charm_tests/tempest/tests.py [hunk 1]
```diff
@@ -89,3 +90,54 @@ def run(self):
         if not keep_workspace or keep_workspace is not True:
             tempest_utils.destroy_workspace(workspace_name, workspace_path)
         return result
+
+
+class TempestTestAgainstKeystoneV2(TempestTestBase):
```
Comment (coreycb) [inline zaza/openstack/charm_tests/tempest/tests.py:95]: I'm wondering if TempestTestWithKeystoneV2 looks better.

**Review by coreycb — APPROVED** [review]
This looks reasonable to me assuming the tests pass.

**Comment by AurelienLourot** [comment]
@coreycb I'm doing a force-push that'll really just rename the test classes both in the code and in the commit message (otherwise we'd be landing a misleading commit message)

**Review by ChrisMacNaughton — APPROVED** [review]
Looks reasonable to me, thanks!

**Pull request merged**